fetch.bs

```diff
@@ -3898,20 +4098,35 @@ these steps:
 
  <li><p>Let <var>actualResponse</var> be null.
 
+ <li><p>Let <var>timingInfo</var> be <var>fetchParams</var>'s<a for="fetch params">timing
+ info</a>.
+
  <li>
   <p>If <var>request</var>'s <a>service-workers mode</a> is "<code>all</code>", then:
 
   <ol>
    <li><p>Let <var>requestForServiceWorker</var> be a <a for=request>clone</a> of
    <var>request</var>.
 
+   <li><p>Set <var>fetchParams</var>'s <a for="fetch params">timing info</a>'s
+   <a for="fetch timing info">worker start time</a> to the <a for=/>unsafe shared current time</a>.
+
    <li><p>Set <var>response</var> to the result of invoking <a for=/>handle fetch</a> for
    <var>requestForServiceWorker</var>. [[!HTML]] [[!SW]]
 
    <li>
     <p>If <var>response</var> is not null, then:
 
     <ol>
+     <li><p>Set <var>fetchParams</var>'s <a for="fetch params">timing info</a>'s
+     <a for="fetch timing info">response start time</a> to the
+     <a for=/>unsafe shared current time</a>.
+
+     <li><p>Set <var>timingInfo</var>'s <a for="fetch timing info">encoded body size</a> and
+     <a for="fetch timing info">decoded body size</a> to the result of calling
+     <a for="header list">extract a length</a> from <var>response</var>'s
+     <a for=response>header list</a>.
```

Comment:
Encoded and decoded body size should map to the actual bytes transferred on the wire, not the Content-Length header (e.g. maybe no Content-Length was sent, maybe it's wrong, etc). Also Content-Length would give you one of those sizes, but not both.

In HTTP-Network fetch you increment them in step 18.1.1.1.3.

If we can't do the same for the SW flow, it'd be better to handwave those parts for now than to set them based on content length.

Otherwise, could we hang those variables on the Response, and then use those in case the SW is reusing a Response to `respondWith()`? 

/cc @wanderview @mfalken

Reply:
Looking closely, cached responses don't get these values at all (they're zeroed). Changed the spec to reflect that and added a note.